src/tokens/selectors.ts

```diff
@@ -321,47 +321,64 @@ export const tokensWithTokenBalanceSelector = createSelector(
   }
 )
 
-export const swappableTokensByNetworkIdSelector = createSelector(
+// sort by balance USD (DESC) then name (ASC), tokens without a priceUsd
+// are pushed last, sorted by name (ASC)
+const sortSwapTokens = (token1: TokenBalance, token2: TokenBalance) => {
+  // Sort by USD balance first (higher balances first)
+  const token1UsdBalance = token1.balance.multipliedBy(token1.priceUsd ?? 0)
+  const token2UsdBalance = token2.balance.multipliedBy(token2.priceUsd ?? 0)
+  if (token1UsdBalance.gt(token2UsdBalance)) return -1
+  if (token1UsdBalance.lt(token2UsdBalance)) return 1
+
+  // Sort by token balance if there is no priceUsd (higher balances first)
+  const balanceCompare = token2.balance.comparedTo(token1.balance)
+  if (balanceCompare) {
+    return balanceCompare
+  }
+
+  // Sort tokens without priceUsd and balance at bottom of list
+  if (token1.priceUsd === null || token2.priceUsd === null) {
+    // If both prices are null, sort alphabetically by name
+    if (!token1.priceUsd && !token2.priceUsd) {
+      return token1.name.localeCompare(token2.name)
+    }
+    // Otherwise, sort such that the token with a non-null price comes first
+    return token1.priceUsd === null ? 1 : -1
+  }
+
+  // Lastly, sort by name
+  return token1.name.localeCompare(token2.name)
+}
+
+export const swappableFromTokensByNetworkIdSelector = createSelector(
   (state: RootState, networkIds: NetworkId[]) => tokensListSelector(state, networkIds),
   (tokens) => {
     const appVersion = deviceInfoModule.getVersion()
-    return (
-      tokens
-        .filter(
-          (tokenInfo) =>
-            tokenInfo.isSwappable ||
-            (tokenInfo.minimumAppVersionToSwap &&
-              !isVersionBelowMinimum(appVersion, tokenInfo.minimumAppVersionToSwap))
+    return tokens
+      .filter((tokenInfo) => {
+        return (
+          tokenInfo.isSwappable ||
+          (tokenInfo.minimumAppVersionToSwap &&
+            !isVersionBelowMinimum(appVersion, tokenInfo.minimumAppVersionToSwap)) ||
+          tokenInfo.balance.gt(0)
```

Comment:
this appear to be the only difference between the two selectors. what do you think about extracting the shared logic into a separate selector?

Reply:
i noticed that in [some places](https://github.com/valora-inc/wallet/blob/845274bbf8d662bd4e0b039bed76c578d3641892/src/tokens/selectors.ts#L320) within this file we consider non-zero balance to be greater than `TOKEN_MIN_AMOUNT`. should we follow this pattern here as well?

Reply:
thank you for the suggestions! makes sense, i applied both your suggestions :)